Mathlib/Algebra/Group/Defs.lean

```diff
@@ -86,6 +89,8 @@ infixl:65 " +ᵥ " => HVAdd.hVAdd
 infixl:65 " -ᵥ " => VSub.vsub
 infixr:73 " • " => HSMul.hSMul
 
+macro_rules | `($x • $y) => `(leftact% HSMul.hSMul $x $y)
```

Comment:
I assume this line is there to propage those coercions to `y` instead of `x`?
In any case, it would be good to document why this line is here.
And please mention this behaviour in the commit message as well.

Reply:
Isn't "propagate *to* `y`" backwards? My understanding is that by default Lean propagates coercions inwards, but this `%leftact` and `%binop` stuff reverses the behavior and propagates them outwards...

Reply:
It makes it so that it participates in the expression tree elaborator on the `y` argument. It computes a type that all the leaf expressions can be coerced to and then inserts coercions where needed. All the basic algebraic operators participate, and the effect is that coercion can propagate into `y` rather than getting stuck around the whole smul expression.

Reply:
There's an example in the RFC https://github.com/leanprover/lean4/issues/2854

Reply:
@kmill Would you mind writing that little documentation above this line?

Reply:
@jcommelin Done. I updated the changes to the HSMul documentation too (I forgot there was anything there, and now I see why you phrased your question the way you did).

Reply:
I've also added more detail in the PR description.